airflow/jobs.py

```diff
@@ -687,6 +695,16 @@ def _execute(self):
         dagbag = models.DagBag(self.subdir, sync_to_db=True)
         executor = self.executor = dagbag.executor
         executor.start()
+
+        # grab orphaned tasks and make sure to reset their state
+        active_runs = DagRun.find(
+            state=State.RUNNING,
+            external_trigger=False,
+            session=session
+        )
+        for dr in active_runs:
+            self._reset_state_for_orphaned_tasks(dr, session=session)
```

Comment:
It shouldnt break a sweat on a couple of thousand. Where exactly is it slow? In the db or in Python?